packages/host/app/components/editor/go.gts

```diff
@@ -219,20 +221,64 @@ export default class Go extends Component<Signature> {
     return file.write(content);
   }
 
-  async saveSingleCardDocument(json: any) {
+  async saveJSONInstance(json: any) {
     let realmPath = new RealmPaths(this.cardService.defaultURL);
-    let url = realmPath.fileURL(
+    let cardUrl = realmPath.fileURL(
       this.args.openFiles.path!.replace(/\.json$/, ''),
     );
 
     try {
-      await this.cardService.saveCardDocument(json, url);
-      await this.loadCard.perform(url);
+      await this.patchCardDocument(json, cardUrl);
+      await this.loadCard.perform(cardUrl);
     } catch (e) {
       console.log('Failed to save single card document', e);
     }
   }
 
+  // NOTE: do not use this function outside of this component unless you are serializing a file
+  // this has been copied from runtime/realm.ts
+  private async fileSerialization(
+    doc: LooseSingleCardDocument,
+    relativeTo: URL,
+  ): Promise<LooseSingleCardDocument> {
+    let card = (await this.cardService.createFromSerialized(
+      doc.data,
+      doc,
+      relativeTo,
+    )) as Card;
+    let data: LooseSingleCardDocument = await this.cardService.serializeCard(
+      card,
+    ); // this strips out computeds
+    delete data.data.id; // the ID is derived from the filename, so we don't serialize it on disk
+    delete data.included;
+    for (let relationship of Object.values(data.data.relationships ?? {})) {
+      delete relationship.data;
+    }
+    return data;
+  }
```

Comment:
I have copied this with the intent of encapsulating fileSerialization to this component

Reply:
Purposely, kept it as private function in runtime-common